routing/turns.cpp

```diff
@@ -51,9 +51,88 @@ static_assert(g_turnNames.size() == static_cast<size_t>(TurnDirection::Count),
 
 namespace routing
 {
-namespace turns
+// UniNodeId -------------------------------------------------------------------
+bool UniNodeId::operator==(UniNodeId const & rhs) const
+{
+  if (m_type != rhs.m_type)
+  {
+    ASSERT(false, ("Comparing UniNodeId of different type. m_type", m_type, "rhs.m_type:", rhs.m_type));
+    return false;
+  }
+
+  switch (m_type) {
+  case Type::Osrm: return m_nodeId == rhs.m_nodeId;
+  case Type::Mwm: return m_featureId == rhs.m_featureId
+        && m_segId == rhs.m_segId && m_forward == rhs.m_forward;
+  }
+}
+
+bool UniNodeId::operator<(UniNodeId const & rhs) const
 {
+  if (m_type != rhs.m_type)
+  {
+    ASSERT(false, ("Comparing UniNodeId of different type. m_type", m_type, "rhs.m_type:", rhs.m_type));
```

Comment:
Mathematically, you don't need this assertion, as your operator< is properly defined on all possible arguments.